### DISCUSSION TITLE
Plans August 2022

### DISCUSSION THREAD

**Discussion by thisisnic** [thisisnic/shinyarrowpi#8] [Ideas]
25 mins + 5 mins Q&A

Take the design process we went through naturally to come up with the architecture and include that in the talk.

Overall thing is that this is a proof-of-concept for a design.

Messages:
1. you can simplify project architecture using Arrow
2. you can do this on the Pi which is awesome because you can use this for sensors and things
3. Arrow + Pi is a great combo because there is limited memory
4. Arrow + Shiny is a great combo for doing fast data manipulation
5. r4pi stuff - local monitoring shiny app

notes:
* monitor what's happening locally with 1 instance but combine them all for a global Shiny app (not what we are doing but we can totally talk about this)

local Shiny app - descriptive analytics - plots etc, what's happened in the past. 1 per sensor.

prod Shiny app - place where we do predictive/prescriptive analytics (e.g. pollution measurement at different times of year, does pollution move around etc). basically accumulating all the stuff from the different sensors.
1 overall.

==============

- shiny app building can go alongside talk writing

What's the plan?
1. abstract
iterate on possible abstracts with mark on the hread there

2. the talk itself
part 1 - mark; sensors and stuff
part 2 - nic; arrow and stuff

plan - write our bits, move them round, profit

**Comment by thisisnic** [comment]
Title: Combining Raspberry Pi, Shiny, and Apache Arrow for Simplifying IoT Project Architecture

Raspberry Pis allow data to be collected from sensors in different location without the need for specialist hardware or expensive machinery. You can supercharge projects like these by taking advantage of Arrow's capabilities for analysing larger-than-memory data directly on the Pi, made even easier by using R4Pi to get everything set up.  Even more exciting is that when you aggregate the data across multiple instances, combining Arrow and Shiny means that you don't have to worry about having too much data to load into your Shiny app.  In this talk we present a PoC for an IoT project and show all of the things I just said.

**Reply by thisisnic** [reply]
@sellorm Here is my shitty first draft!  Realised we needed a title too.  Feel free to totally disregard this and write something new!

**Reply by sellorm** [reply]
Looks great (apart from that last sentence! 🤣)

Is there a nice way to shorten the title though?

* Raspberry Pi, Apache Arrow and Shiny for great good
* Raspberry Pi, Apache Arrow and Shiny for IoT
* Apache Arrow in IoT
* Firing an Arrow into the internet of things
* IoT with Apache Arrow and the Raspberry Pi

It's fine as it is, it just feels a bit long.

**Reply by thisisnic** [reply]
~"and show all of the things I just said."~  "demonstrating how this would work in practice".?

I like "Raspberry Pi, Apache Arrow and Shiny for IoT" and "Firing an Arrow into the internet of things" best.  Torn between them as the first is a nice 'telling you what this is about' box-ticker, whereas the second has wordplay and wordplay is always fun.  You pick!

Sure you don't wanna make any more changes to the abstract itself?  Can have more R4Pi stuff in there - I just didn't know what to say.

Once we have a "good enough to show to other people" draft, I might take it to the VD devrel team to see if they can help make it "pop".

**Reply by thisisnic** [reply]
@sellorm OK, this is the next version, let me know what you think:

Firing an Arrow into the internet of things: combining the power of Arrow, Raspberry Pi & Shiny

Raspberry Pis are very cheap and very small computers, and one of their uses is allowing data to be collected from sensors in different locations without the need for specialist hardware or expensive machinery.  You can supercharge projects like these by taking advantage of Arrow's capabilities for analysing larger-than-memory data directly on the Pi, made even easier by using the pre-built R packages provided by the R4Pi project to get everything set up. Even more exciting is that when you aggregate the data across multiple instances, combining Arrow and Shiny means that you don't have to worry about overloading your Shiny app with too much data. In this talk we present a proof-of-concept for an IoT project and demonstrate how this would work in practice.

**Reply by sellorm** [reply]
This is perfect, Nic - thanks!
